lib/plugins/aws/package/compile/events/msk/index.js

```diff
@@ -40,6 +40,79 @@ class AwsCompileMSKEvents {
       additionalProperties: false,
       required: ['arn', 'topic'],
     });
+
+    const secretsManagerArnRegex =
+      'arn:[a-z-]+:secretsmanager:[a-z0-9-]+:\\d+:secret:[A-Za-z0-9/_+=.@-]+';
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'kafka', {
+      type: 'object',
+      properties: {
+        accessConfigurations: {
+          type: 'array',
+          items: {
+            type: 'object',
+            minProperties: 1,
+            maxProperties: 1,
+            properties: {
+              basicAuth: {
+                anyOf: [
+                  {
+                    type: 'string',
+                    pattern: secretsManagerArnRegex,
+                  },
+                  { type: 'object' },
+                ],
+              },
+              vpcSubnet: {
+                anyOf: [
+                  {
+                    type: 'string',
+                    pattern: 'subnet-[a-z0-9]+',
+                  },
+                  { type: 'object' },
```

Comment:
Just to confirm, this is here to support potential support for intrinsic functions with mandatory `!Sub` to construct proper value, correct?

Reply:
Yes, exactly.